### PR TITLE
Implement filter slots by time

### PR DESCRIPTION
**Pull request by mntss**

### CODE REVIEW

File: src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
 
         $scope.JobProfileEnum = JobProfileEnum;
         $scope.currentJobProfile = JobProfileEnum.dev;
 
         $scope.days = tdprDateService.getCurrentWeek();
         $scope.slotsTimes = slotsTimes;
+        $scope.selectSlotsTimes = slotsTimes;
         $scope.persons = persons;
+        $scope.endTime = slotsTimes[slotsTimes.length - 1].endTime;
+        $scope.startTime = slotsTimes[0].startTime;
```
Comment (rlewan) [inline src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js:14]: Are we confident that slotTimes array will always contain at least 2 elements? If not some checks should be implemented.

File: src/main/resources/assets/js/application/recruiter/filters/tdprSlotsByTimeFilter.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
-define(['application/recruiter/tdprRecruiterModule'], function (tdprRecruiterModule) {
+define(['angular', 'application/recruiter/tdprRecruiterModule'], function (angular, tdprRecruiterModule) {
 
     tdprRecruiterModule.filter('slotsByTime', function() {
 
         return function(input, startTime, endTime) {
             var filteredInput = [];
             input.forEach(function(el) {
-                if (el.startTime >= startTime && el.endTime <= endTime) {
+                if ((el.startTime >= startTime || angular.isUndefined(startTime)) && (el.endTime <= endTime || angular.isUndefined(endTime))) {
```
Comment (rlewan) [inline src/main/resources/assets/js/application/recruiter/filters/tdprSlotsByTimeFilter.js:8]: Perhaps it's worth to update unit tests to include the undefined scenario?

File: src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js (hunk 1 above)
Comment (rlewan) [inline src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js:14]: Undefined is handled in the filter -- never mind then :)

File: src/main/resources/assets/js/application/recruiter/filters/tdprSlotsByTimeFilter.js (hunk 2 above)
Comment (mntss) [inline src/main/resources/assets/js/application/recruiter/filters/tdprSlotsByTimeFilter.js:8]: Done

**Pull request merged**